src/content/blog/2025-03-25-tus-node-server-v2.0.0.md

```diff
@@ -0,0 +1,185 @@
+[`require(esm)`](https://joyeecheung.github.io/blog/2024/03/18/require-esm-in-node-js/)
+from CommonJS. An incredible achievement and finally relieving maintainers from
+choosing whether to publish CommonJS, ESM, or both.
+
+With this release we bump the minimum required Node.js version from >=16
+to >=20.19.0, the specific release that backported `require(esm)`.
+
+While we don’t depend on any ESM-only packages yet, it does give us the
+flexibility to do so should we need it, ensuring we can pick the best tools to
+offer the best uploading experience.
+
+### Running anywhere where JavaScript runs
+
+tus Node.js 2.0.0 can now run in all meta frameworks (such as Next.js, Nuxt,
+React Router, SvelteKit, etc) and all Node.js compatible runtime environments
+(AWS Lambda, Cloudflare (as long as the store does not depend on `node:fs`),
+Bun, Deno Deploy, etc). This major version is a rewrite of all handlers to be
+based on `Request` and `Response`, as it’s possible to convert Node’s
+request/response objects to those but not the other way around.
+
+Use the new `handleWeb()` method for `Request` based handlers, such as in Bun:
+
+```js
+import { Server } from '@tus/server'
+import { FileStore } from '@tus/file-store'
+
+const tus = new Server({
+  path: '/files',
+  datastore: new FileStore({ directory: './files' }),
+})
+
+Bun.serve({
+  routes: {
+    '/files/*': (req) => tus.handleWeb(req),
+  },
+})
+```
+
+Use the `handle()` method for integrating into traditional Node.js frameworks,
+such as Fastify:
+
+```js
+import fastify from 'fastify'
+import { Server } from '@tus/server'
+import { FileStore } from '@tus/file-store'
+
+const app = fastify()
+const tus = new Server({
+  path: '/files',
+  datastore: new FileStore({ directory: './files' }),
+})
+app.addContentTypeParser(
+  'application/offset+octet-stream',
+  // Leave body untouched
+  (request, payload, done) => done(null),
+)
+app.all('/files', (req, res) => tus.handle(req.raw, res.raw))
+app.all('/files/*', (req, res) => tus.handle(req.raw, res.raw))
+app.listen(3000)
+```
+
+Or run the server standalone:
+
+```js
+import { Server } from '@tus/server'
+import { FileStore } from '@tus/file-store'
+
+const host = '127.0.0.1'
+const port = 1080
+
+const tus = new Server({
+  path: '/files',
+  datastore: new FileStore({ directory: './files' }),
+})
+tus.listen({ host, port })
+```
+
+## What is next?
+
+With this release, the server is in really good shape. Nonetheless some
+improvements might be on the horizon soon.
+
+- Structured logging with log levels.
+- New distributed locks: a S3 locker based on
+  [conditional writes](https://www.morling.dev/blog/leader-election-with-s3-conditional-writes/)
+  and Google Cloud Storage locker (already WIP) with a
+  [similar approach](https://www.joyfulbikeshedding.com/blog/2021-05-19-robust-distributed-locking-algorithm-based-on-google-cloud-storage.html),
+  removing the need for something like Redis or Postgres.
+- Even more stability?
```

Comment:
What do you mean by this? It makes it sounds like tus-node-server is not yet super stable. Maybe expand on this point or leave it out if there's nothing concrete to say.